src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -485,6 +494,17 @@ func Register(
 		}
 	}
 
+	// Application services can register users with no auth type, but require
+	// access token. Differentiate from users who are initially hitting register
+	// without an auth type
+	if r.Auth.Type == "" && req.URL.Query().Get("access_token") == "" {
```

Comment:
Application services can use Auth headers for this endpoint as well I assume.
So this check is "incomplete"

Reply:
Good point, let me add that in.